src/controllers/userController.js

```diff
@@ -37,12 +39,33 @@ const signUp = async (req, res) => {
 
     return response(res, 200, 'success', userData);
   } catch (e) {
-    return response(res, 500, 'error', {
-      errors: e
-    });
+    return response(res, 500, 'error', { errors: e });
+  }
+};
+
+/**
+ * user signin controller
+ * @param {Object} req - server request
+ * @param {Object} res - server response
+ * @returns {Object} - custom response
+*/
+
+const signIn = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+    const user = await findByEmail(email);
+    if (!user) return response(res, 404, 'error', { message: messages.userNotFound });
+    const { password: hash, ...data } = user.dataValues;
+    const passwordsMatch = await comparePasswords(password, hash);
+    if (!passwordsMatch) return response(res, 400, 'error', { message: messages.incorrectPassword });
+    const token = authHelper.generateToken({ id: data.id });
+    return response(res, 200, 'success', { ...data, token });
+  } catch (error) {
+    response(res, 500, 'error', { error });
```

Comment:
Use sinon to test this line stub the User object and findOne method to throw error.

Reply:
this will make your test coverage pass

Reply:
Okay @tejiri4,  looking into it